### PR TITLE
feat!: rename Claude plugin to planbridge@contextbridge; refresh plugins on update

### LINKED ISSUE #49: feat: move Claude Code plugin assets into this repo, rename to planbridge@contextbridge
### What's the problem?

PlanBridge's Claude Code integration is split across two repos:

- **`contextbridge/planbridge`** (this repo) — the `contextbridge` binary, including the `contextbridge hook claude` subcommand the plugin actually invokes, plus the `ClaudeInstaller` that wires Claude Code up via `claude plugin marketplace add` / `claude plugin install`.
- **`contextbridge/claude-plugin`** — `marketplace.json` (the catalog), `plugin.json`, and `hooks/hooks.json` declaring `PermissionRequest:ExitPlanMode → contextbridge hook claude`.

The original reason for the split was that this repo wasn't public — the plugin manifest had to live somewhere users could `claude plugin marketplace add`, so it lived in a separate public repo. That constraint is gone, but two specific files (`plugin.json`, `hooks/hooks.json`) being in the wrong repo still costs us:

- **Drift risk.** `hooks.json`'s command string (`contextbridge hook claude`) lives in one repo and the `hookClaude.ts` implementation lives in the other. Nothing in CI cross-validates them; a rename in one repo silently breaks the other.
- **Two-repo release dance.** Any change touching both the hook contract and the manifest needs coordinated PRs across repos.
- **Discoverability.** A contributor reading this repo can't see how Claude Code actually invokes the CLI without knowing the other repo exists.

While we're in here: the plugin is named `cli`, so its install id is `cli@contextbridge`. That made sense when `cli` was the only thing — it doesn't anymore. The product is PlanBridge, not "the CLI," and once sibling plugins ship (code review, etc.) `cli` becomes meaningless as a distinguishing name.

There's also a deeper problem the move surfaces: today `contextbridge update` swaps the binary and stops there. That works only because the hook contract is `command: "contextbridge hook claude"` — a PATH lookup that picks up the new binary automatically. The moment the plugin id, `hooks.json`, or `plugin.json` `version` changes (this PR or future hook-contract changes), the binary and plugin drift. Worth fixing as part of this PR: couple binary and plugin updates.

### Proposed solution

Two changes, landing together. Take the breaking change on the plugin rename now while the user base is small — much cheaper than doing it later.

**Keep `contextbridge/claude-plugin` as the marketplace catalog.** It's the storefront where users discover all contextbridge plugins (today: PlanBridge; future: code review, etc.). Each plugin's actual source lives in its own product repo. Claude Code's marketplace schema supports cross-repo plugin sources, so this is straightforward.

**Move plugin assets into this repo:**

```
harnessIntegrations/claude/.claude-plugin/plugin.json
harnessIntegrations/claude/hooks/hooks.json
```

(Top-level folder is `harnessIntegrations` so future Codex / other harness assets land as siblings — `harnessIntegrations/claude/`, future `harnessIntegrations/codex/`, etc.)

**Rename the plugin** in the new `plugin.json`: `"name": "cli"` → `"name": "planbridge"`. New install id is `planbridge@contextbridge`.

**Edit `contextbridge/claude-plugin/.claude-plugin/marketplace.json`** to point at this repo via a cross-repo source:

```diff
   "plugins": [
     {
-      "name": "cli",
-      "source": "./cli",
-      "description": "ContextBridge Claude Code integration — opens plan reviews in the browser on ExitPlanMode."
+      "name": "planbridge",
+      "source": { "source": "github", "repo": "contextbridge/planbridge", "path": "harnessIntegrations/claude" },
+      "description": "PlanBridge — opens plan reviews in the browser on ExitPlanMode."
     }
   ]
```

Confirmed locally against `claude plugin validate .` (Claude Code 2.1.132): the nested object form is canonical; the shorthand string `"github:..."` is rejected as invalid input.

The `cli/` directory in `contextbridge/claude-plugin` is deleted.

**Code changes in this repo:**

- `packages/cli/src/harnesses/ClaudeInstaller.ts` — `PLUGIN_ID = 'planbridge@contextbridge'`; add `LEGACY_PLUGIN_ID = 'cli@contextbridge'`. `MARKETPLACE_SOURCE` unchanged.
- `status()` surfaces both new and legacy plugin entries in `managed[]`. `installed` stays gated on the new id only.
- `runInstall(scope)`: if `cli@contextbridge` is at `scope`, uninstall it before installing the new id. Emits one stderr line about the rename.
- `runUninstall(scope)`: also uninstalls `cli@contextbridge` at the target scope when present.
- `update.ts`: after a successful binary swap, walk `ALL_INSTALLERS`, find harnesses with managed state, and spawn `<process.execPath> install <harness-id>` per harness. The new binary's installer migrates legacy and refreshes the plugin contract.
- `release-please-config.json`: add `extra-files` so `harnessIntegrations/claude/.claude-plugin/plugin.json`'s `version` bumps with each binary release.

### Transition path for existing users

`contextbridge update` now couples binary and plugin updates. After the binary swap, `runUpdate` walks installers, skips harnesses with no PlanBridge state, and spawns the new binary's per-harness `install <harness>` for those with state. For Claude with legacy `cli@contextbridge` installed, the new binary's `runInstall`:

1. detects legacy at the target scope via `claude plugin list --json`,
2. runs `claude plugin uninstall cli@contextbridge --scope <scope>`,
3. emits stderr: `PlanBridge plugin renamed from cli@contextbridge to planbridge@contextbridge — migrated automatically.`,
4. runs `claude plugin marketplace add contextbridge/claude-plugin` (idempotent),
5. runs `claude plugin install planbridge@contextbridge --scope <scope>`.

Refresh failure logs a warning and never blocks update success. `contextbridge install` continues to work as the explicit/manual entrypoint and performs the same migration. `contextbridge uninstall` cleans up the legacy plugin too if present. The Codex installer is untouched.

### Alternatives you considered

- **Move everything (including `marketplace.json`) into this repo and archive `contextbridge/claude-plugin`.** Considered and rejected. Keeping the marketplace repo separate is a feature, not vestigial: it's the org-level catalog of all contextbridge plugins. When the next plugin ships, it gets its own product repo and a new entry in the same `contextbridge/claude-plugin/marketplace.json`. Users see one storefront; each product owns its own implementation.
- **Keep the plugin name `cli`** to avoid the breaking change. Cheaper in the short term, but the rename only gets more expensive as the user base grows. Doing it now is the right cost-optimization.
- **Keep the split as-is.** Status quo; rejected because the constraint that drove it is gone and the drift cost grows as more harness plugins land.
- **Banner / stderr nudge for users still on the old binary.** Considered and dropped. They'll get the update eventually, and from then on every future update keeps things current.

### Additional context

Existing layout in `contextbridge/claude-plugin`:

```
.claude-plugin/marketplace.json     # marketplace "contextbridge", plugin source: "./cli"
cli/.claude-plugin/plugin.json      # plugin name "cli", version "0.1.0"
cli/hooks/hooks.json                # PermissionRequest:ExitPlanMode -> "contextbridge hook claude"
```

After this lands:

```
contextbridge/claude-plugin
└── .claude-plugin/marketplace.json     # plugin "planbridge", cross-repo source -> contextbridge/planbridge

contextbridge/planbridge
└── harnessIntegrations/
    └── claude/
        ├── .claude-plugin/plugin.json  # plugin name "planbridge"
        └── hooks/hooks.json            # unchanged contract: contextbridge hook claude
```

Cross-repo plugin source is documented at https://code.claude.com/docs/en/plugin-marketplaces.md — the docs explicitly call out that "a marketplace hosted at `acme-corp/plugin-catalog` can list a plugin fetched from `acme-corp/code-formatter`."

### Implementation

- contextbridge/claude-plugin#1 — moves the marketplace pointer to the new cross-repo source, deletes `cli/`.
- contextbridge/planbridge#52 — adds the plugin assets, renames the plugin id, couples binary and plugin updates.

**Sequencing**: merge `claude-plugin#1` first (catalog flips), then `planbridge#52`, then the planbridge release-please release PR.

### PR DESCRIPTION
**Pull request by blimmer**
## Summary

Move PlanBridge's Claude Code plugin assets (`plugin.json`, `hooks.json`) from `contextbridge/claude-plugin` into this repo at `harnessIntegrations/claude/`, rename the plugin id from `cli@contextbridge` to `planbridge@contextbridge`, and couple binary + plugin updates so `contextbridge update` re-runs install for harnesses that already have PlanBridge wired up. Closes #49.

## Review focus

- **`runUpdate` post-swap refresh** in `packages/cli/src/commands/update.ts` walks `ALL_INSTALLERS`, skips harnesses with `status.managed.length === 0`, and spawns `<process.execPath> install <harness-id>` per harness with state. Refresh failure logs a warning and never blocks update success. Spawning `process.execPath` (not the old `runUpdate`'s in-process logic) is deliberate — the child runs the _new_ binary's installer code, which knows about the rename and updates/migrates Claude plugin state automatically.
- **`ClaudeInstaller.status()`** now surfaces both `planbridge@contextbridge` and `cli@contextbridge` entries in `managed[]`, but `installed` stays gated on the new id only. This is what makes the refresh filter naturally migrate legacy users without a separate "do I have legacy state?" abstraction.
- **`runInstall`/`runUninstall` legacy cleanup** at the target scope only — see `ClaudeInstaller.ts`. Tests cover the cross-scope edge case (legacy at project, install at user → leaves legacy alone). Existing `planbridge@contextbridge` installs now use `claude plugin update` so `contextbridge update` pulls the latest plugin version.
- **`release-please-config.json` `extra-files`** keeps `harnessIntegrations/claude/.claude-plugin/plugin.json`'s `version` in sync with binary releases.

**Sequencing**: depends on [contextbridge/claude-plugin#1](https://github.com/contextbridge/claude-plugin/pull/1) merging first so the new plugin id resolves in the marketplace catalog. Then this PR; then the release-please release PR.

## Commits

- [`b2807dd`](https://github.com/contextbridge/planbridge/commit/b2807dda623dbbec2855ab190c1cda6656ae486f) — package the Claude Code plugin integration assets in this repo
- [`a382919`](https://github.com/contextbridge/planbridge/commit/a382919da258e6139ed01143284de55fb31bea5b) — rename the Claude plugin to `planbridge@contextbridge`
- [`9494b1f`](https://github.com/contextbridge/planbridge/commit/9494b1f00310762977a2d65cf1ee1998bbd0cf41) — refresh managed harness integrations after `contextbridge update`

### CODE REVIEW

File: harnessIntegrations/claude/.claude-plugin/plugin.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (blimmer) [inline harnessIntegrations/claude/.claude-plugin/plugin.json:1]: This moves from github.com/contextbridge/claude-plugin (see https://github.com/contextbridge/claude-plugin/pull/1). That repo is now just a catalog of all ContextBridge claude plugins (right now, just PlanBridge, but could expand to other products in the future).

I'm doing this migration now, which will make https://github.com/contextbridge/planbridge/issues/51 easier to implement in a single place.

File: packages/cli/src/commands/install.test.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ describe('runInstall', () => {
       'user',
     ]);
     expect(pluginInstall).toHaveLength(1);
-    expect(pluginInstall[0]?.args).toEqual(['plugin', 'install', 'cli@contextbridge', '--scope', 'user']);
+    expect(pluginInstall[0]?.args).toEqual(['plugin', 'install', 'planbridge@contextbridge', '--scope', 'user']);
```
Comment (blimmer) [inline packages/cli/src/commands/install.test.ts]: I'm renaming this to match up. If we don't feel strongly, we can leave it as `cli` - feedback welcome.

File: packages/cli/src/commands/update.test.ts [hunk 3]
```diff
@@ -189,4 +202,193 @@ describe('runUpdate', () => {
...
+    commandRunner.on(process.execPath, ['install', 'claude']).resolves();
+
+    await runUpdate(context);
+
+    const spawnCall = commandRunner.calls.find((c) => c.cmd === process.execPath);
+    expect(spawnCall).toBeDefined();
+    expect(spawnCall?.args).toEqual(['install', 'claude', '--scope', 'user']);
+  });
+
+  it('refreshes Claude at project scope when the new plugin is installed at project scope', async () => {
```
Comment (blimmer) [inline packages/cli/src/commands/update.test.ts:255]: Things could get a little bit weird if the old named plugin is installed in various projects, but that didn't feel like something I wanted to work around.

File: packages/cli/src/harnesses/ClaudeInstaller.ts [hunk 4]
```diff
@@ -42,25 +43,43 @@ export class ClaudeInstaller extends ScopedHarnessInstaller {
...
     await runPluginCommand(ctx, binaryName, 'marketplace add', [
       'marketplace',
       'add',
       MARKETPLACE_SOURCE,
       '--scope',
       scope,
     ]);
-    await runPluginCommand(ctx, binaryName, 'install', ['install', PLUGIN_ID, '--scope', scope]);
+    if (hasCurrentAtScope) {
+      await runPluginCommand(ctx, binaryName, 'update', ['update', PLUGIN_ID, '--scope', scope]);
```
Comment (blimmer) [inline packages/cli/src/harnesses/ClaudeInstaller.ts]: Claude will update plugins on some cadence, but we want to force it to be up-to-date (e.g., when we introduce new harness features).

File: release-please-config.json [hunk 5]
```diff
@@ -6,6 +6,13 @@
       "package-name": "contextbridge",
       "include-component-in-tag": false,
       "changelog-path": "CHANGELOG.md",
+      "extra-files": [
+        {
+          "type": "json",
+          "path": "harnessIntegrations/claude/.claude-plugin/plugin.json",
+          "jsonpath": "$.version"
```
Comment (blimmer) [inline release-please-config.json:13]: This is a handy feature that'll keep the claude plugin released in-step with the binary.

File: harnessIntegrations/claude/.claude-plugin/plugin.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  "name": "planbridge",
+  "description": "PlanBridge — opens plan reviews in the browser on ExitPlanMode.",
+  "version": "0.2.0",
+  "author": {
+    "name": "ContextBridge AI, Inc."
+  },
+  "homepage": "https://plan.contextbridge.ai",
+  "repository": "https://github.com/contextbridge/planbridge",
+  "license": "MIT",
+  "keywords": ["contextbridge", "planbridge", "planning", "review", "hooks", "ExitPlanMode"]
```
Comment (jcarver989) [inline harnessIntegrations/claude/.claude-plugin/plugin.json]: Is `ExitPlanMode` in the keywords intentional? Just curious since it sticks out against the others.

File: packages/cli/src/commands/install.test.ts (hunk 2 above)
Comment (jcarver989) [reply packages/cli/src/commands/install.test.ts]: Think `planbridge` will make more sense given the marketing site and product name

File: packages/cli/src/commands/install.test.ts [hunk 7]
```diff
@@ -49,15 +49,17 @@ describe('runInstall', () => {
       .resolves(marketplaceListResult([{ name: 'contextbridge' }]));
     commandRunner
       .on(CLAUDE_BINARY, ['plugin', 'list', '--json'])
-      .resolves(pluginListResult([{ id: 'cli@contextbridge', scope: 'user' }]));
+      .resolves(pluginListResult([{ id: 'planbridge@contextbridge', scope: 'user' }]));
```
Comment (jcarver989) [inline packages/cli/src/commands/install.test.ts]: Seems like we should use the constant that's defined elsewhere in this PR across these tests.

File: packages/cli/src/commands/update.ts [hunk 8]
```diff
@@ -57,6 +60,53 @@ export function registerUpdate(ctx: CliContext, program: Command): void {
     });
 }
 
+// After the binary has been swapped, re-run the new binary's per-harness install
+// for any harness with existing PlanBridge state. Keeps installed plugins in
+// sync with the binary across renames and hook-contract changes. Skips harnesses
+// the user never wired up; never blocks update success on a refresh failure.
+async function refreshInstalledHarnesses(ctx: CliContext): Promise<void> {
+  const { commandRunner, logger } = ctx;
+  for (const installer of ALL_INSTALLERS) {
```
Comment (jcarver989) [inline packages/cli/src/commands/update.ts:68]: Probably just me being me, but I'd `flatMap` here (using empty array in the failure case) as it'd avoid the `let` binding + re-assignment.

File: packages/cli/src/commands/update.ts [hunk 9]
```diff
@@ -57,6 +60,53 @@ export function registerUpdate(ctx: CliContext, program: Command): void {
...
+    let refreshScopes: string[];
+    try {
+      const status = await installer.status(ctx);
+      refreshScopes = getRefreshScopes(status.managed);
+    } catch (err) {
+      logger.warn({ err, harness: installer.descriptor.id }, 'post-update harness refresh failed');
+      continue;
+    }
+
+    for (const scope of refreshScopes) {
```
Comment (jcarver989) [inline packages/cli/src/commands/update.ts]: Noting that this is sequential, assuming this is intentional since concurrent install commands might cause issues?

**Review by jcarver989 — APPROVED** [review]

File: harnessIntegrations/claude/.claude-plugin/plugin.json (hunk 6 above)
Comment (blimmer) [reply harnessIntegrations/claude/.claude-plugin/plugin.json]: Ah, just copied this over verbatim from the other repo - you're right though, not very helpful. Let me update.

File: packages/cli/src/commands/update.ts (hunk 8 above)
Comment (blimmer) [reply packages/cli/src/commands/update.ts:68]: Sure - I'll take a more functional approach throughout.

File: packages/cli/src/commands/update.ts (hunk 9 above)
Comment (blimmer) [reply packages/cli/src/commands/update.ts]: Yep, it's intentional. `claude plugin install` mutates shared config under `~/.claude` (and Codex hook installs touch `~/.codex/hooks.json`), so parallel installs across scopes/harnesses would race on those files. Sequential is also fine performance-wise — the post-update refresh fan-out is small (one per managed scope per installer), and `stdio: 'inherit'` already streams progress to the user.
